src/api/models/revisions.py

```diff
@@ -0,0 +1,207 @@
+import uuid
+
+from django.contrib.gis.db import models
+
+
+class Revision(models.Model):
+    table_name = models.CharField(max_length=50, db_index=True, editable=False)
+    record_id = models.UUIDField(db_index=True, editable=False)
+    project_id = models.UUIDField(null=True, db_index=True, editable=False)
+    profile_id = models.UUIDField(null=True, db_index=True, editable=False)
+    revision_num = models.IntegerField(null=False, default=1)
+    updated_on = models.DateTimeField(editable=False)
+    deleted = models.BooleanField(default=False, editable=False)
+
+    class Meta:
+        db_table = "revision"
+        unique_together = ("table_name", "record_id")
+
+    def __str__(self):
+        return f"[{self.revision_num}] {self.table_name} {self.record_id}"
+
+
+# -- TRIGGER SQL --
+
+forward_sql = """
+    CREATE SEQUENCE revision_seq_num START 1;
```

Comment:
Just wondering... once we create this sequence, everything -- all users everywhere, offline, etc. -- depends on it (since it's not per-date/table/project/whatever), and I think adjustments or rollbacks might be difficult. Since we're doing this by trigger anyway, is it worth considering just doing the nextval logic internally, i.e. `max(revision_num) + 1`?

Reply:
I left the sequence but am setting start value after the sequence has been created

```
CREATE SEQUENCE revision_seq_num START 1;
SELECT setval('revision_seq_num', (SELECT MAX(revision_num) FROM revision));
```